From 0e0618a21a3f1e13b2bbe1e5ca28dceda578fa0a Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Sun, 12 Mar 2017 14:09:06 +0200
Subject: [PATCH 0609/1051] crypto: inside-secure: store cell-index value

This patch stores the cell-index which is read from
the device-tree.

Change-Id: I61b1d9862801d1564599e3d79133594a51fcb265
Signed-off-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/inside-secure/safexcel.c | 6 ++++++
 drivers/crypto/inside-secure/safexcel.h | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index 7e71043457a6..fdce3243c808 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -1002,6 +1002,12 @@ static int safexcel_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->base);
 	}
 
+	ret = of_property_read_u32(dev->of_node, "cell-index", &priv->id);
+	if (ret) {
+		dev_err(dev, "failed to read cell-index property\n");
+		return ret;
+	}
+
 	priv->clk = devm_clk_get(&pdev->dev, NULL);
 	ret = PTR_ERR_OR_ZERO(priv->clk);
 	/* The clock isn't mandatory */
diff --git a/drivers/crypto/inside-secure/safexcel.h b/drivers/crypto/inside-secure/safexcel.h
index 65624a81f0fd..7d10bac2f895 100644
--- a/drivers/crypto/inside-secure/safexcel.h
+++ b/drivers/crypto/inside-secure/safexcel.h
@@ -606,6 +606,8 @@ struct safexcel_crypto_priv {
 	atomic_t ring_used;
 
 	struct safexcel_ring *ring;
+
+	int id;
 };
 
 struct safexcel_context {
-- 
2.17.1

